packages/core/src/browser/shell/view-contribution.ts

```diff
@@ -108,8 +108,23 @@ export abstract class AbstractViewContribution<T extends Widget> implements Comm
             };
             await shell.addWidget(widget, widgetArgs);
         } else if (args.toggle && area && shell.isExpanded(area) && tabBar.currentTitle === widget.title) {
-            // The widget is attached and visible, so close it (toggle)
-            await this.closeView();
+            // The widget is attached and visible, so collapse the containing panel (toggle)
+            switch (area) {
+                case 'left':
+                case 'right':
+                    shell.collapsePanel(area);
+                    break;
+                case 'bottom':
+                    // Don't collapse the bottom panel if it's currently split
+                    if (shell.bottomAreaTabBars.length === 1) {
+                        shell.collapsePanel('bottom');
+                    }
+                    break;
+                default:
+                    // The main area cannot be collapsed, so close the widget
+                    await this.closeView();
```

Comment:
I like the behavior much more with this pull-request 👍 

I'm wondering in the case of the `main-panel`, should we default to a `no-op` instead of closing the view.

In all other areas widgets are simply collapsed and are recoverable by users (they understand that toggling means hiding), but for the main-area it is untrue as the widget is explicitly closed (something which they might not actually want to happen/like).

Reply:
Usually a view shouldn't be put in the main area. But if for some reason someone decides to use AbstractViewContribution for a main area widget, the generated command can still toggle the widget by adding and removing it from that area.

If we decide for a no-op in the main area case, we should change the command label `'Toggle ' + this.viewLabel + ' View'` to `'Show ' + this.viewLabel + ' View'` for such cases.

Reply:
i think we can go with close for now, I've never seen anyone doing it, let's wait for a real issue from users?